### DISCUSSION TITLE
Performance

### DISCUSSION THREAD

**Discussion by fabnavigator** [zwave-js/node-red-contrib-zwave-js#193] [Q&A]
I found that it is necessary to add in a rate limit delay node when I'm sending messages to the zwave controller node. Is this expected, or am I doing something wrong? Do incoming messages go into a queue and the next one is sent after the preceeding one is finished?

**Comment by marcus-j-davies** [comment]
Hi,

We are not aware of needing rate limiting. What makes you believe you need to throttle? Unless you are stressing the controller itself like asking it send to 1000 network commands in 1 sec?

Any message processing outward or inward is taken care of with zwave-js and/or the usb stick it's  self. Node red doesn't have a say here. 

We bridge directly to zwave-js. So it should perform like any other front end.

**Reply by fabnavigator** [reply]
@marcus-j-davies I think everything is okay. Maybe I was working thgough some other issues when I thoughtt he delay was necessary. I've removed it, and have rewritten a bunch of code, and everything looks good. I've completed my migration from home assistant.
